.github/workflows/pull_request.yaml

```diff
@@ -2,57 +2,84 @@ name: Pull Request
 on:
   pull_request:
     types: [opened, reopened, synchronize]
+
 permissions:
   contents: read
   pull-requests: write
+
 env:
   BASE_ENVIRONMENT: 'development'
   ENVIRONMENT_TYPE: 'development'
-  HUMCTL_VERSION: '0.23.2'
   ENVIRONMENT_ID: pr-${{ github.event.number }}
   ENVIRONMENT_NAME: PR-${{ github.event.number }}
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+      
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GH_TOKEN }}
+
       - uses: humanitec/setup-cli-action@v1
         with:
-          version: ${{ env.HUMCTL_VERSION }}
+          version: ${{ vars.HUMCTL_VERSION }}
+
       - name: Create Humanitec Env
+        if: ${{ github.event.action == 'opened' || github.event.action == 'reopened' }}
         run: |
-            humctl create environment ${{ env.ENVIRONMENT_ID }} \
-              --token ${{ secrets.HUMANITEC_TOKEN }} \
-              --org ${{ vars.HUMANITEC_ORG }} \
-              --app ${{ vars.APP_NAME }} \
-              --name ${{ env.ENVIRONMENT_NAME }} \
-              -t ${{ env.ENVIRONMENT_TYPE }} \
-              --from ${{ env.BASE_ENVIRONMENT }} \
-              || true
-      - uses: score-spec/setup-score@v2
-        with:
-          file: score-humanitec
-          version: ${{ env.SCORE_HUMANITEC_VERSION }}
+          humctl create environment ${{ env.ENVIRONMENT_ID }} \
+            --token ${{ secrets.HUMANITEC_TOKEN }} \
+            --org ${{ vars.HUMANITEC_ORG }} \
+            --app ${{ vars.APP_NAME }} \
+            --name ${{ env.ENVIRONMENT_NAME }} \
+            -t ${{ env.ENVIRONMENT_TYPE }} \
+            --from ${{ env.BASE_ENVIRONMENT }}
+        
+      - name: Build and push image
+        id: build_push
+        run: |
+          # Make sure repo name is all lowercase for docker push
+          GH_REPOSITORY=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
+
+          TAG=$GH_REPOSITORY:${{ env.ENVIRONMENT_ID }}
+          IMAGE=ghcr.io/$TAG
+
+          # Set this label to connect the package to the GitHub repo to make it public
+          docker build . --tag $IMAGE  \
+            --label "org.opencontainers.image.source=https://github.com/$GH_REPOSITORY"
+          docker push $IMAGE
+          echo "IMAGE=$IMAGE" >> $GITHUB_OUTPUT
       - name: Run Score
         run: |
           humctl score deploy \
```

Comment:
Let's add an explicit message/comment like this below?
```
--message "$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "${{ github.event.pull_request.commits_url }}?per_page=10" | jq -r .[-1].commit.message)" \
```

Reply:
I added a message showing the latest git commit message. This works (I tested it) without the curl to the GH API, preventing a dependency to GitHub for any adopters. I needed to change the `actions/checkout@v4` to use

```yaml
      - uses: actions/checkout@v4
        with:
          # Checkout using the head commit to read commit message further down
          ref: ${{ github.event.pull_request.head.sha }}
```

based on [this discussion](https://github.com/orgs/community/discussions/28474#discussioncomment-3334578).